runtime/recipe/handle.js

```diff
@@ -183,12 +183,27 @@ class Handle {
     }
   }
 
+  // Options:
+  // - showUnresolved: if true, includes unresolved handle details if
+  //   handle is in fact unresolved and they're available.
+  // - stripManifestSessionFromImmediate: if true and the handle is
+  //   an immediate, strips the session id prefix from the handle id.
   toString(nameMap, options) {
     // TODO: type? maybe output in a comment
     let result = [];
     result.push(this.fate);
     if (this.id) {
-      result.push(`'${this.id}'`);
+      // TODO(wkorman): `Manifest._processRecipe` has a TODO currently to mark a
+      // handle as immediate. Until we've a cleaner way to identify, we just
+      // scrutinize the handle id for a clear marker. A sample immediate handle
+      // id looks like:
+      // '!130690574120708:manifest:./manifest.manifest::0:immediateB'
+      let outputId = this.id.toString();
+      if (options && options.stripManifestSessionFromImmediate &&
```

Comment:
I thought adding an option was a not crazy tweak to resolve this, but two notes to consider:
- we could consider (separately would be fine IMO) reworking `Recipe.digest` to not depend on `toString`, which I would have assumed is mainly only for debugging and logging purposes.
- `Recipe.toString` itself turns around and does a `replace` on the result of this method, as `result.push(handle.toString(nameMap, options).replace(/^|(\n)/g, '$1  '));`. I'm not sure what that replace is aiming at, it would benefit from a comment, but we may want to incorporate that as part of the aforementioned potential cleanup.